Mimir.Worker/Services/MongoDbStore.cs

```diff
@@ -66,11 +66,19 @@ public async Task UpdateLatestBlockIndex(long blockIndex)
         await MetadataCollection.BulkWriteAsync(new[] { updateModel });
     }
 
-    public async Task<long> GetLatestBlockIndex()
+    public async Task<long?> GetLatestBlockIndex()
     {
         var filter = Builders<BsonDocument>.Filter.Eq("_id", "SyncContext");
-        var doc = await MetadataCollection.FindSync(filter).FirstAsync();
-        return doc.GetValue("LatestBlockIndex").AsInt64;
+        try
+        {
+            var doc = await MetadataCollection.FindSync(filter).FirstAsync();
+            return doc.GetValue("LatestBlockIndex").AsInt64;
+        }
+        catch (InvalidOperationException e)
+        {
+            Console.WriteLine(e);
```

Comment:
Since using `Microsoft.Extensions.Logging.ILogger<T>`, it is better to use `ILogger<T>.LogError` instead.

```suggestion
            _logger.LogError(e, e.Messsage);
            // Or
            _logger.LogError(e, "Some Message (e.g., 'Failed to get LatestBlockIndex from metadata collection')");
```

Reply:
Thanks to catch it!